packages/components/src/slot-fill/bubbles-virtually/slot.tsx

```diff
@@ -13,12 +17,26 @@ import { useMergeRefs } from '@wordpress/compose';
 /**
  * Internal dependencies
  */
+import { View } from '../../view';
 import SlotFillContext from './slot-fill-context';
+import type { WordPressComponentProps } from '../../ui/context';
+import type { SlotComponentProps } from '../types';
 
 function Slot(
-	{ name, fillProps = {}, as: Component = 'div', ...props },
-	forwardedRef
+	props: WordPressComponentProps<
+		Omit< SlotComponentProps, 'bubblesVirtually' >,
+		'div'
+	>,
+	forwardedRef: ForwardedRef< any >
 ) {
+	const { name, fillProps = {}, as: Component = 'div', ...restProps } = props;
+
+	// `children` is not allowed. However, if it is passed,
+	// it will be displayed as is, so remove `children`.
+	if ( 'children' in restProps ) {
+		delete restProps.children;
+	}
```

Comment:
This mutation isn't great. Can we figure out some other way to do it? Using another object that omits children for example.

Reply:
Once this PR is organized, create a separate PR for this section only.

Reply:
fixed: https://github.com/WordPress/gutenberg/pull/53272.